src/stores/DAOEntityListStore.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import type { DAO, QueryOptions } from 'brewskey.js-api';
+
+import { action, observable } from 'mobx';
+import flattenArray from 'array-flatten';
+import DAOApi, { LoadObject } from 'brewskey.js-api';
+import { createRange } from '../utils';
+
+export type Row<TEntity> = {|
+  key: number,
+  value: LoadObject<TEntity>,
+|};
+
+class DAOEntityListStore<TEntity> {
+  _baseQueryOptions: QueryOptions = {};
+  _dao: DAO<TEntity, *>;
+  _daoSubscriptionID: string;
+  _pageSize: number;
+
+  @observable _queryOptionsList: Array<QueryOptions> = [];
+
+  @observable rows: Array<Row<TEntity>> = [];
+
+  constructor(
+    dao: DAO<TEntity, *>,
+    queryOptions?: QueryOptions = {},
+    pageSize?: number = 20,
+  ) {
+    this._baseQueryOptions = queryOptions;
+    this._dao = dao;
+    this._pageSize = pageSize;
+
+    this._daoSubscriptionID = this._dao.subscribe(this._computeRows);
+    this.reset();
+  }
+
+  dispose = () => {
+    this._dao.unsubscribe(this._daoSubscriptionID);
+  };
+
+  @action
+  fetchNextPage = () => {
+    const currentQuery = this._queryOptionsList[
+      this._queryOptionsList.length - 1
+    ];
+
+    // prevent fetch when we reached end of the list
+    const currentQueryLoadObject = this._dao.fetchMany(currentQuery);
```

Comment:
I don't think this code will work in all cases.

If the user scrolls faster than the sections can load, `fetchNextPage` won't work initially.  I guess that is OK if there is a spinner or something at the end of the list while in this state.